compiler-core/src/language_server/tests/action.rs

```diff
@@ -845,6 +846,113 @@ pub type Wibble { Wibble(arg1: Int, arg2: Int) }
         .is_none())
 }
 
+#[test]
+fn test_assign_unused_result() {
+    assert_code_action!(
+        2,
+        ASSIGN_UNUSED_RESULT,
+        "
+pub fn main() {
+    Ok(0)
+    Nil
+}
+"
+    );
+}
+
+#[test]
+fn test_assign_unused_result_in_block() {
+    assert_code_action!(
+        3,
+        ASSIGN_UNUSED_RESULT,
+        "
+pub fn main() {
+    {
+        Ok(0)
+        Nil
+    }
+    Nil
+}
+"
+    );
+}
+
+// #[test]
+// fn test_assign_unused_result_on_block_start() {
```

Comment:
do you have any suggestions on how to test code actions that only apply when on a specific line

Reply:
Sorry, what do you mean? The first argument is the line

Reply:
Sorry, I should make it clearer. This is what I am trying to achieve(similar to what `rust-analyzer` does):
```gleam
pub fn main() {
//  ↓ will show the trigger to turn it into: let _ = { Ok(0)\nOk(0)\n };
    {
        Ok(0) // will only show one action to turn it into: let _= Ok(0) but no the one for outer block
        Ok(0)
    }
//  ↑ will show the trigger to turn it into: let _ = { Ok(0)\nOk(0) }
    Nil
}
```
and the `engine_response` function use `range: Range::new(Position::new(0, 0), Position::new(line + 1, 0))` for code actions which will fail if I want to test if the code action only show up on a specific line like the above example. I hope it is clear this time🙏.

Reply:
The existing behaviour seems OK to me?

Reply:
Roger, I made it only for the start and end line because when I was testing this feature, showing 2 actions confused me a little bit. If you feel okay with that, I will do it.
<img width="992" alt="image" src="https://github.com/user-attachments/assets/4ddf8251-bc22-47a0-9d2e-7371e543e0dc">

Reply:
I think it should only show for the inner-most one